community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/DefaultNodeCursor.java

```diff
@@ -195,8 +198,15 @@ else if ( hasChanges && txs.nodeIsDeletedInThisTx( next ) )
                 next++;
                 setInUse( false );
             }
+            else if ( lastStoreReference + 1 == next )
+            {
+                lastStoreReference = next;
+                read.nodeAdvance( this, pageCursor );
+                next++;
+            }
             else
             {
+                lastStoreReference = next;
                 read.node( this, next++, pageCursor );
```

Comment:
Would it be possible to assign `lastStoreReference` after the `next++` lines, initialise `lastStoreReference` to `-2`, aka. `NO_ID`, and change `else if ( lastStoreReference + 1 == next )` to `else if ( lastStoreReference == next )` ? Maybe then `lastStoreReference` could be called `nextScanReference`. Perhaps that would make the code clearer.